src/lib.rs

```diff
@@ -1,3 +1,109 @@
+//! you view inside a signal so that you can react to updates and update the `View`. Signals are reactive values that can be read from and written to.
+//!
+//! To affect the layout and rendering of your component, you will need to send a state update to your component with [Id::update_state](id::Id::update_state)
+//! and then call [UpdateCx::request_layout](context::UpdateCx::request_layout) to request a layout which will cause a repaint.
+//!
+//! To share state between components, you can simply pass down a signal to your children. Here's a contrived example:
+//!
+//!```rust,no_run
+//! // super contrived example showing how to pass down state to children.
+//!
+//! struct Parent<V> {
+//!     id: Id,
+//!     text: ReadSignal<String>,
+//!     child: V,
+//! }
+//!
+//! // Creates a new parent view with the given child.
+//! fn parent<V>(new_child: impl FnOnce(ReadSignal<String>) -> V) -> Parent<impl View>
+//! where
+//!     V: View + 'static,
+//! {
+//!     let text = create_rw_signal(cx.scope, "World!".to_string());
+//!     // share the signal between the two children
+//!     let (id, child) = AppContext::new_id_with_child(stack(|| (text_input(text)), new_child(text.read_only()));
+//!     Parent { id, text, child }
+//! }
+//!
+//! impl<V> View for Parent<V>
+//! where
+//!     V: View,
+//! {
+//! // implementation omitted for brevity
+//! }
+//!
+//! struct Child {
+//!     id: Id,
+//!     label: Label,
+//! }
+//!
+//! /// Creates a new child view with the given state (a read only signal)
+//! fn child(text: ReadSignal<String>) -> Child {
+//!     let (id, label) = AppContext::new_id_with_child(|| label(move || format!("Hello, {}", text.get()));
+//!     Child { id, label }
+//! }
+//!
+//! impl View for Child {
+//!   // implementation omitted for brevity
+//! }
+//!
+//! // Usage
+//! fn main() {
+//!     floem::launch(parent(child));
+//! }
+//!
+//! ```
+//!
+//! Global state TBD
+//!
+//! # Styling
+//! Floem sizing and positioning layout system is based on the flexbox model using Taffy as the layout engine.
+//! Styles are divided into two parts:
+//! [`ComputedStyle`]: A style with definite values for most fields.  
+//!
+//! [`Style`]: A style with [`StyleValue`]s for the fields, where `Unset` falls back to the relevant
+//! field in the [`ComputedStyle`] and `Base` falls back to the underlying [`Style`] or the
+//! [`ComputedStyle`].
+//!
+//!
+//! ## Render loop and update lifecycle
+//! TBD: event -> update -> layout -> paint.
+//!
+//! potentially discuss update messages, deferred update messages, animations
+//!
+//! ## Notable invariants and tolerances
+//! - There can be only one root `View`
+//! - Only one view can be active at a time.
+//! - Only one view can be focused at a time.
+//! - Any other important invariants?
+//!
+//! # Terminology
+//! - **Active view**: The view that can receive mouse events even if the mouse is outside its bounds.
```

Comment:
I don't fully understand what the difference between active and focused is.

Reply:
Pointer events will only be sent to the ```active``` View. It is useful when you drag things, e.g. the scroll bar, you set the scroll bar active after pointer down, then when you drag, the ```PointerMove``` will always be sent to the View, even if your mouse is outside of the view. 

And ```focused``` is for sending keyboards events. Keyboard events are only sent to focused view.